php_mysqlnd_azure.c

```diff
@@ -28,39 +28,52 @@
 
 ZEND_DECLARE_MODULE_GLOBALS(mysqlnd_azure)
 
-/* {{{ PHP_GINIT_FUNCTION */
-static PHP_GINIT_FUNCTION(mysqlnd_azure)
-{
-#if defined(COMPILE_DL_MYSQLND_AZURE) && defined(ZTS)
-	ZEND_TSRMLS_CACHE_UPDATE();
-#endif
-	mysqlnd_azure_globals->enabled = 0;
-	mysqlnd_azure_globals->redirectCache = NULL;
-}
-/* }}} */
 
-/* {{{ PHP_GSHUTDOWN_FUNCTION */
-static PHP_GSHUTDOWN_FUNCTION(mysqlnd_azure)
+/* {{{ OnUpdateEnableRedirect */
+static ZEND_INI_MH(OnUpdateEnableRedirect)
 {
-	if (mysqlnd_azure_globals->redirectCache) {
-		zend_hash_destroy(mysqlnd_azure_globals->redirectCache);
-		mnd_pefree(mysqlnd_azure_globals->redirectCache, 1);
-		mysqlnd_azure_globals->redirectCache = NULL;
-	}
+    if ((ZSTR_LEN(new_value) == 9 && strcasecmp("preferred", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 1 && strcasecmp("2", ZSTR_VAL(new_value)) == 0)) {
+
+        MYSQLND_AZURE_G(enableRedirect) = REDIRECT_PREFERRED;
+
+    } else if ((ZSTR_LEN(new_value) == 2 && strcasecmp("on", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 3 && strcasecmp("yes", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 4 && strcasecmp("true", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 1 && strcasecmp("1", ZSTR_VAL(new_value)) == 0)) {
+
+        MYSQLND_AZURE_G(enableRedirect) = REDIRECT_ON;
+
+    } else {
+
+        MYSQLND_AZURE_G(enableRedirect) = REDIRECT_OFF;
```

Comment:
I am not familiar with API. Just curious when this code path will be hit? If customer wants to set it 'on' to test his configuration, but set it by 'on1' by mistake, will the option changed to unexpected 'off' automatically?

Reply:
Yes, but that's customary for PHP.